src/Datadog.Trace/TraceContext.cs

```diff
@@ -49,7 +48,7 @@ public TraceContext(IDatadogTracer tracer)
 
         public void AddSpan(Span span)
         {
-            lock (_lock)
+            lock (_spans)
```

Comment:
Same comment as earlier about using a data item as the lock object.

Reply:
Having the separate `_lock` incurs an additional object allocation per trace, which can translate to hundreds of objects per second. In the interest of allocating as little memory as possible, I suggest drop `_lock` and lock on the `_spans` field instead.

Reply:
ok, I'm good with that.